File: Sliding Window/Medium/1493. Longest Subarray of 1's After Deleting One Element/longest_subarray_of_1s_after_deleting_one_element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int ans = 0;
        int zeros = 0;

        for (int left = 0, right = 0; right < nums.size(); right++)
        {
            if (nums[right] == 0)
                zeros++;
            while (zeros == 2)
            {
                if (nums[left] == 0)
                    zeros--;
                left++;
            }
            ans = max(ans, right - left);
        }

        return ans;
    }
};

File: Sliding Window/Medium/1493. Longest Subarray of 1's After Deleting One Element/longest_subarray_of_1s_after_deleting_one_element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        ans = 0
        zeros = 0

        left = 0
        for right in range(len(nums)):
            if nums[right] == 0:
                zeros += 1
            while zeros == 2:
                if nums[left] == 0:
                    zeros -= 1
                left += 1
            ans = max(ans, right - left)

        return ans
